File: src/main/java/me/lockie/coopersmpwinter/items/CustomItemManager.java
package me.lockie.coopersmpwinter.items;

import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;

public class CustomItemManager {

    public ArrayList<CustomItem> customItems = new ArrayList<>();

    public ItemStack getCustomItemById(String id) {
        for (CustomItem customItem : this.customItems)
            if (customItem.id.equals(id))
                return customItem.item;
        return null;
    }

    public CustomItemManager() {
        customItems.add(new SnowShovel());
        customItems.add(new HotChocolate());
    }

    public void registerCustomItems() {
        this.customItems.forEach(customItem -> {
            customItem.registerItem();
            customItem.registerRecipe();
        });
    }

}

File: src/main/kotlin/me/lockie/coopersmpwinter/items/CustomItemManager.kt
package me.lockie.coopersmpwinter.items

import org.bukkit.inventory.ItemStack
import java.util.function.Consumer

class CustomItemManager {
    var customItems = ArrayList<CustomItem>()
    fun getCustomItemById(id: String): ItemStack? {
        for (customItem in customItems) if (customItem.id == id) return customItem.item
        return null
    }

    init {
        customItems.add(SnowShovel())
        customItems.add(HotChocolate())
    }

    fun registerCustomItems() {
        customItems.forEach(Consumer { customItem: CustomItem ->
            customItem.registerItem()
            customItem.registerRecipe()
        })
    }
}
